cmd/net.go

```diff
@@ -37,6 +37,16 @@ var localIP4 = mustGetLocalIP4()
 // IPv6 address of local host.
 var localIP6 = mustGetLocalIP6()
 
+// IsIPv4 - check if address is IPv4 by counting number of ':' which should be 0 or 1
+func IsIPv4(address string) bool {
+	return strings.Count(address, ":") < 2
+}
+
+// IsIPv6 - check if address is IPv4 by counting number of ':' which should be 2 or more
+func IsIPv6(address string) bool {
+	return strings.Count(address, ":") >= 2
```

Comment:
Same here it should be simple as `ip.To4() == nil` is IPv6.

Reply:
I agree that counting colon's assumes that the input is already constrined to a valid IPv4 or IPv6 address.  For example, `IsIpV6("x:y:z:::::ismyfavorite")` would return true.

I originally used this implementation because it was simply so that I could quickly check if brackets needed to be added in `updateDomainIps()` and I was relying on other logic to deal with truely invalid inputs.

I thought of `net.ParseIP()` but it is is very pedantic about how an IPv6 address is formatted and wasn't consistent with what `net.SplitHostPort()` expects.

```
net.ParseIP("::FFFF:0A00:0001") # returns IP
net.ParseIP("[::FFFF:0A00:0001]") # returns nil

net.SplitHostPort("::FFFF:0A00:0001") # thorws error that too many colons in address
net.SplitHostPort("[::FFFF:0A00:0001]") # throws expected missing port error
```
I agree a more robust implementation would be good.  I don't believe I can simply change `ip.To4() == nil` because the input "::FFFF:0A00:0001" when given to `ParseIP` will create an IPv4 address.  While this makes sense in some contexts since the address is an IPv4 address expressed in IPv6 format, in this situation I want to know about the string format.  

I'll change the implementation to this.

```
func IsIPv6FormattedAddress(address string) bool {
               if strings.Count(address, ":") < 2 {
            return false
        }

	if !strings.HasPrefix(address, "[") {
	    address = "[" + address + "]"
	}
	
        // SplitHostPort requires IPv6 addresses include brackets
        host, _, err := net.SplitHostPort(address)
	if err != nil {
		if strings.Contains(err.Error(), "missing port in address") {
			host = address;
		}
	}
	
	if strings.HasPrefix(host, "[") && strings.HasSuffix(host, "]") {
		host = host[1 : len(host)-1]
	}
	

        // because we forced address to have at least 2 colons in it above
        // if the address parses and To16() succeeds we know this is an IPv16
        // formatted address
	ip := net.ParseIP(host)
	if (ip != nil && ip.To16() != nil) {
	    return true
	}
	
	return false
}
```